smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpRpcProtocolGenerator.java

```diff
@@ -135,9 +136,17 @@ private void generateOperationSerializer(GenerationContext context, OperationSha
                        + "): Promise<$T> {", "}", methodName, inputType, requestType, () -> {
             writeRequestHeaders(context, operation);
             boolean hasRequestBody = writeRequestBody(context, operation);
+            boolean hasHostPrefix = operation.hasTrait(EndpointTrait.class);
+
+            if (hasHostPrefix) {
```

Comment:
Nice! This is a good example of where duplicating code is better than trying to make an abstraction.